compiler/aarch64/codegen/BinaryEvaluator.cpp

```diff
@@ -30,33 +30,67 @@
 #include "il/Node_inlines.hpp"
 #include "infra/Bit.hpp"
 
-static TR::Register *addOrSubInteger(TR::Node *node, TR::CodeGenerator *cg)
+static TR::Register *
+addOrSub32Or64Integer(TR::Node *node, TR::CodeGenerator *cg)
    {
    TR::Node *firstChild = node->getFirstChild();
    TR::Register *src1Reg = cg->evaluate(firstChild);
    TR::Node *secondChild = node->getSecondChild();
-   TR::Register *trgReg = cg->allocateRegister();
+   TR::Register *trgReg = NULL;
+
+   if (1 == firstChild->getReferenceCount())
+      {
+      trgReg = cg->evaluate(firstChild);
```

Comment:
You could just use `src1Reg` here rather than calling the `evaluate` function again.

Reply:
Ok. I'll fix it up.